packaging/distributions-dependencies/build.gradle.kts

```diff
@@ -98,7 +98,8 @@ dependencies {
         api(libs.groovyXml)             { version { strictly(libs.groovyVersion) }}
         api(libs.gson)                  { version { strictly("2.10") }}
         api(libs.h2Database)            { version { strictly("2.2.220") }}
-        api(libs.hamcrestCore)          { version { strictly("1.3"); because("2.x changes the API") }}
+        api(libs.hamcrest)              { version { strictly("3.0") }}
+        api(libs.hamcrestCore)          { version { strictly("3.0") }}
```

Comment:
Is `libs.hamcrestCore` used anywhere anymore? I think we should replace all our usages of `libs.hamcrestCore` with `libs.hamcrest` and remove `hamcrestCore` from `ExternalModulesExtension`

Reply:
Interesting, I didn't realize hamcrest-core actually existed. It contains a simple jar with contents
![image](https://github.com/user-attachments/assets/e7bd3e47-8189-46a4-a953-34833b7cd126)

and a single dependency on `org.hamcrest:hamcrest:3.0`.

This wouldnt work otherwise.

`capabilities.json` only comes in to place when _both_ the old coordinates and the new coordinates are on the same classpath -- so if the user declares a dependency on both. It protects us from having duplicate classes on the classpath.

This is probably fine.



Though what I find most odd is that users can still write `libs.hamcrestCore` even though you'd never want to declare a dependency on it. We only want it here so we can force a version upgrade and not so users can declare a dependency on it